src/components/DataSource/independent.ts

```diff
@@ -0,0 +1,42 @@
+import { ValueRef } from '@holoflows/kit/es'
+import Services from '../../extension/service'
+import { PersonaArrayComparer } from '../../utils/comparer'
+import { MessageCenter } from '../../utils/messages'
+import { Persona } from '../../database'
+import { useValueRef } from '../../utils/hooks/useValueRef'
+import { PluginMessageCenter } from '../../plugins/PluginMessages'
+import { currentEthereumNetworkSettings } from '../../plugins/Wallet/network'
+import { WalletRecord, ERC20TokenRecord } from '../../plugins/Wallet/database/types'
+
+const independentRef = {
+    myPersonasRef: new ValueRef<Persona[]>([], PersonaArrayComparer),
+    // TODO: typeonly import
+    walletTokenRef: new ValueRef<[(WalletRecord & { privateKey: string })[], ERC20TokenRecord[]]>([[], []]),
+}
+
+{
+    const ref = independentRef.myPersonasRef
+    query()
```

Comment:
You might want to wrap them in the `sideEffects().then`

Reply:
I think it's ok (at least for now) as no one needs to import this except the options page.

Reply:
It can prevent code to be executed in the test environment

Reply:
If it fails in the test environment, it should fail.

Reply:
it should not run in the test environment 👀

Reply:
@Jack-Works @SunriseFox update on this?

Reply:
I request changes on this 🙈

Reply:
Update?

Reply:
@SunriseFox @Jack-Works